### PR TITLE
update Babel to 7.7.0 and enable error recovery

### PR DESCRIPTION
**Pull request by thorn0**
- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -393,6 +393,7 @@
         "<!-- prettier-ignore -->\\n(`{3,})\\w*\\n[\\s\\S]+?\\1",
         "\\[@\\w+?\\]",
         "\\[`\\w+`\\]",
-        "ve+r+y+(long\\w+)?"
+        "ve{2,}r{2,}y",
+        "ve+r+y+long\\w*"
     ]
-}
+}
```
[the remark this answers is not in the document]
Comment (sosukesuzuki) [reply cspell.json]: Probably it due to https://github.com/prettier/prettier/pull/6684#discussion_r336745585

File: src/language-js/parser-babylon.js [hunk 2]
```diff
@@ -17,6 +17,7 @@ function babelOptions(extraOptions, extraPlugins) {
       allowReturnOutsideFunction: true,
       allowSuperOutsideMethod: true,
       allowUndeclaredExports: true,
+      errorRecovery: true,
```
[the remark this answers is not in the document]
Comment (thorn0) [reply src/language-js/parser-babylon.js]: Justification?

[the remark this answers is not in the document]
Comment (brodybits) [reply src/language-js/parser-babylon.js]: > What do you call "default configuration of Babel"?

I meant default behavior: what Babel would do if we do not add `errorRecovery: true` here.

[the remark this answers is not in the document]
Comment (lydell) [reply src/language-js/parser-babylon.js]: @brodybits Have you read this? https://babeljs.io/blog/2019/11/05/7.7.0#parser-error-recovery-10363-https-githubcom-babel-babel-pull-10363

Babel mentions Prettier as a good use case for error recovery. Their example is very good.

**Comment by brodybits** [comment]
> I think we're about to drop a new patch version, so might want to hold off a sec!

I think the purpose of this proposal is to avoid a breaking change in the new patch version.

Alternative could be to roll @babel/parser back to say 7.2.0. (I would favor including this PR in the patch release.)

**Comment by existentialism** [comment]
@brodybits I meant that Babel will be dropping a patch version soon

File: src/language-js/parser-babylon.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (thorn0) [reply src/language-js/parser-babylon.js]: @brodybits Sorry, but your argument sounds like we should use the default option values just for the sake of their defaultness.

**Comment by thorn0** [comment]
@existentialism This patch release doesn't include `@babel/parser`, right?

**Comment by lydell** [comment]
@existentialism Is this good to go, or should we wait for something?

**Comment by lydell** [comment]
We’ll make a new PR if needed.

**Comment by existentialism** [comment]
@lydell yep! sorry for delay, missed that there wasn't a parser update

**Pull request merged**